hermes-console-vue/src/composables/constraints/use-constraints/useConstraints.ts

```diff
@@ -29,6 +48,144 @@ export function useConstraints(): UseConstraints {
     return constraints.value?.subscriptionConstraints ?? {};
   });
 
+  const doUpsertTopicConstraint = async (
+    topicName: string,
+    constraint: Constraint,
+  ) => {
+    const notificationsStore = useNotificationsStore();
+
+    try {
+      await upsertTopicConstraint(topicName, constraint);
+
+      await notificationsStore.dispatchNotification({
+        title: useGlobalI18n().t(
+          'notifications.constraints.topic.created.success',
+          {
+            topicName,
+          },
+        ),
+        text: '',
+        type: 'success',
+      });
+    } catch (e) {
+      await notificationsStore.dispatchNotification({
+        title: useGlobalI18n().t(
+          'notifications.constraints.topic.created.failure',
+          {
+            topicName,
+          },
+        ),
+        text: (e as Error).message,
+        type: 'error',
+      });
+    } finally {
+      fetchConstraints();
```

Comment:
WDYT about instead of finally block and fetching data again, just returning boolean with operation result and depending on it refreshing page with 'router.go(0)'?

Reply:
good idea, changed to page refresh